src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -765,30 +827,28 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                                 // into targetReg. Next, do the predicated operation on the targetReg and last,
                                 // use "sel" to select the active lanes based on mask, and set inactive lanes
                                 // to falseReg.
-
                                 assert(targetReg != embMaskOp2Reg);
                                 assert(HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinEmbMask.id));
 
-                                GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, embMaskOp1Reg);
-
-                                emitInsHelper(targetReg, maskReg, embMaskOp2Reg);
+                                emitInsMovPrfxHelper(targetReg, maskReg, embMaskOp1Reg, embMaskOp2Reg);
                             }
 
                             GetEmitter()->emitIns_R_R_R_R(INS_sve_sel, emitSize, targetReg, maskReg, targetReg,
                                                           falseReg, opt);
-                            break;
                         }
                         else if (targetReg != embMaskOp1Reg)
                         {
                             // embMaskOp1Reg is same as `falseReg`, but not same as `targetReg`. Move the
                             // `embMaskOp1Reg` i.e. `falseReg` in `targetReg`, using "unpredicated movprfx", so the
                             // subsequent `insEmbMask` operation can be merged on top of it.
-                            GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, falseReg);
+                            emitInsMovPrfxHelper(targetReg, maskReg, falseReg, embMaskOp2Reg);
```

Comment:
same here. this was earlier generating unpredicated version, but now is generated predicated one.

Reply:
`emitInsMovPrfxHelper()` generates an unpredicated movprfx:
`GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, reg1, reg3);`

Reply:
However.... looks like HEAD is wrong:

```
                            else
                            {
                                // If the instruction just has "predicated" version, then move the "embMaskOp1Reg"
                                // into targetReg. Next, do the predicated operation on the targetReg and last,
                                // use "sel" to select the active lanes based on mask, and set inactive lanes
                                // to falseReg.

                                assert(targetReg != embMaskOp2Reg);
                                assert(HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinEmbMask.id));

                                GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, embMaskOp1Reg);

                                emitInsHelper(targetReg, maskReg, embMaskOp2Reg);
                            }
```

That is generating a unpredicated movprfx followed by a predicated instruction.

(This was spotted by my newest changes in #106184)